neo3/wallet/wallet.py

```diff
@@ -0,0 +1,198 @@
+
+
+# A sample schema, like what we'd get from json.load()
+schema = {
+    "type": "object",
+    "properties": {
+        "path": {"type": "string"},
+        "name": {"type": "string"},
+        "scrypt": {"$ref": "#/$defs/scrypt_parameters"},
+        "accounts": {
+            "type": "array",
+            "items": {"$ref": "#/$defs/account"},
+            "minItems": 0,
+        },
+        "extra": {"type": ["object", "null"],
+                  "properties": {},
+                  "additionalProperties": True
+                  },
+    },
+    "required": ["path", "name", "scrypt", "accounts", "extra"],
+    "$defs": {
+        "account": {
+            "type": "object",
+            "properties": {
+                "address": {"type": "string"},
+                "label": {"type": "string"},
+                "is_default": {"type": "boolean"},
+                "lock": {"type": "boolean"},
+                "key": {"type": "string"},
+                "contract": {"type": ""},
+                "extra": {"type": ["object", "null"],
+                          "properties": {},
+                          "additionalProperties": True}
+            },
+            "required": ["address", "label", "is_default", "lock", "key", "contract", "extra"]
+
+        },
+        "scrypt_parameters": {
+            "type": "object",
+            "properties": {
+                "n": {"type": "integer"},
+                "r": {"type": "integer"},
+                "p": {"type": "integer"},
+                "length": {"type": "integer"}
+            },
+            "required": ["n", "r", "p"]
+        }
+    }
+}
+
+
+class Wallet(IJson):
+
+    _wallet_version = '3.0'
+
+    def __init__(self,
+                 path: str,
+                 version: str,
+                 scrypt: ScryptParameters,
+                 name: Optional[str] = None,
+                 accounts: List[Account] = None,
+                 extra: Optional[Dict[Any, Any]] = None):
+        """
+        Args:
+            path: the JSON's path
+            name: a label that the user has given to the wallet
+            version: the wallet's version, must be equal or greater then 3.0
+            scrypt: a ScryptParameters object which describes the parameters of the SCrypt algorithm used for encrypting
+                    and decrypting the private keys in the wallet.
+            accounts: an array of Account objects which describe the details of each account in the wallet.
+            extra: an object that is defined by the implementor of the client for storing extra data. This field can be
+                   None.
+        """
+
+        self.path = path
+        self.name = name
+        self.version = version
+        self.scrypt = scrypt
+        self.accounts = accounts if accounts is not None else []
+        self.extra = extra
+
+    @classmethod
+    def default(cls, path: str, name: Optional[str] = None) -> Wallet:
```

Comment:
I have an idea here; how about we make `default()` create a new wallet were
- `path` = current directory `.`
- `name` =  `wallet.json`
- rest remains as is (default scrypt, no accounts, no extra)

If they want more control, they can call `__init__`. What's your opinion?

Reply:
This seems fine to me

Just one more thing: if the user would call `__init__`, what about changing `scrypt` with a default value as well?
By setting its default as `scrypt: ScryptParameters = ScryptParameters()`, so the user isn't forced to set the script parameters by themself.